File: src/main/java/io/zeitwert/fm/portfolio/service/api/dto/PortfolioEvaluationResult.java
package io.zeitwert.fm.portfolio.service.api.dto;

import java.util.List;

import io.zeitwert.fm.building.service.api.dto.EvaluationBuilding;
import io.zeitwert.fm.building.service.api.dto.EvaluationElement;
import io.zeitwert.fm.building.service.api.dto.EvaluationPeriod;
import lombok.Builder;
import lombok.Data;

@Data()
@Builder
public class PortfolioEvaluationResult {

	private Integer id;
	private String name;
	private String description;
	private String accountName;

	private List<EvaluationBuilding> buildings;

	private List<EvaluationElement> elements;

	private Integer startYear;
	private List<EvaluationPeriod> periods;

}

File: fm-domain/src/main/java/io/zeitwert/fm/portfolio/api/dto/PortfolioEvaluationResult.kt
package io.zeitwert.fm.portfolio.api.dto

import io.zeitwert.fm.building.api.dto.EvaluationBuilding
import io.zeitwert.fm.building.api.dto.EvaluationElement
import io.zeitwert.fm.building.api.dto.EvaluationPeriod

data class PortfolioEvaluationResult(
	val id: Int,
	val name: String,
	val description: String,
	val accountName: String,

	val buildings: List<EvaluationBuilding>,

	val elements: List<EvaluationElement>,

	val startYear: Int,
	val periods: List<EvaluationPeriod>,
)
